ls1intum/Artemis
# `Development`: Match Java distributions for the build and execution of Artemis

### Checklist
#### General



### Motivation and Context


This is a follow-up PR from the following PR:
- #5721 

@b-fein mentioned that the [openjdk image](https://hub.docker.com/_/openjdk) is deprecated. The last update for the [previous image](https://hub.docker.com/layers/library/openjdk/17-jdk-slim/images/sha256-779635c0c3d23cc8dbab2d8c1ee4cf2a9202e198dfc8f4c0b279824d9b8e0f22) has been released six months ago and therefore not received any security updates since.
The new eclipse-temurin:17-jre image is also smaller by avoiding including the full JDK when not needed for running the application (531MB vs 720MB before).

As the GitHub workflows uses Zulu as an Java distro we wanted to match all of Artemis Java Distros for building, testing and execution in this PR. We decided for an extra PR to be able to test this separately and being able to revert easily if problems should occur.

### Description

This changes/matches the Java distribution for the whole Artemis project to Temurin instead of the plain OpenJDK or/and Zulu.

Temurin is already being used in the [docker image which is used for the Bamboo builds and tests](https://github.com/ls1intum/docker-jdk-node-yarn) and seems to be a good and well maintained open-source java distro.

### Steps for Testing

- build and execute the related Dockerfiles
- check this branch for the github workflow if there are errrors

### Review Progress





#### Code Review
#### Manual Tests
